src/remap/function/ipv6_to_ip.rs

```diff
@@ -0,0 +1,116 @@
+use std::net::IpAddr;
+
+use remap::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct Ipv6ToIp;
+
+impl Function for Ipv6ToIp {
+    fn identifier(&self) -> &'static str {
+        "ipv6_to_ip"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            accepts: |v| matches!(v, Value::Bytes(_)),
+            required: true,
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required_expr("value")?;
+
+        Ok(Box::new(Ipv6ToIpFn { value }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct Ipv6ToIpFn {
+    value: Box<dyn Expression>,
+}
+
+impl Ipv6ToIpFn {
+    #[cfg(test)]
+    fn new(value: Box<dyn Expression>) -> Self {
+        Self { value }
+    }
+}
+
+impl Expression for Ipv6ToIpFn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        let ip = {
+            let bytes = self.value.execute(state, object)?.try_bytes()?;
+            String::from_utf8_lossy(&bytes)
+                .parse()
+                .map_err(|err| format!("unable to parse IP address: {}", err))?
+        };
+
+        match ip {
+            IpAddr::V4(addr) => Ok(Value::from(addr.to_ipv6_mapped().to_string())),
+            IpAddr::V6(addr) => match addr.to_ipv4() {
+                Some(addr) => Ok(Value::from(addr.to_string())),
+                None => Err(format!("IPV6 address {} is not compatible with IPV4", addr).into()),
+            },
+        }
+    }
+
+    fn type_def(&self, state: &state::Compiler) -> TypeDef {
+        self.value
+            .type_def(state)
+            .fallible_unless(value::Kind::Bytes)
```

Comment:
@JeanMertz Is this correct? The function will error if the parameter is not a valid ip address. Does this function need to capture this?